2_prep.R

```diff
@@ -0,0 +1,160 @@
+source('2_prep/src/munge_nmls.R')
+source('2_prep/src/munge_meteo.R')
+source('2_prep/src/munge_inouts.R')
+source('2_prep/src/harmonize_dates.R')
+source('2_prep/src/visualize_inputs.R')
+
+p2 <- list(
+
+  #### Munge model inputs ####
+
+  # Set the list of reservoirs to proceed with. As currently coded, we extract the reservoir IDs from the nml list,
+  # but this code may be overridden by a vector or subsetting code to reduce the number of models being processed in
+  # p2 and subsequent phases
+  tar_target(
+    p2_reservoir_ids,
+    readRDS(p1_ltmp_nml_list.rds) %>%
+      purrr::map_chr('site_id') %>%
+      {setNames(., purrr::map_chr(p1_nml_edits, 'site_name')[.])} %>% # add layperson reservoir names to the vector
+      .[] # .[c(1,1,2)] # subset/customize the list if desired
+  ),
+  # Identify the set of meteo files corresponding to the [potentially subsetted] p2_reservoir_ids
+  tar_files(
+    p2_meteo_files,
+    {
+      depends <- p1_meteo_files
```

Comment:
I'm confused by this syntax here. Do you have to specify `p1_meteo_files` as a depends for the `p2_meteo_files` target because you're not calling a function within the call to `tar_files()`, and therefore aren't passing `p2_meteo_files` directly as an argument to that function? Also - are the curly brackets simply necessary b/c you have more than one operation specified as the second argument to `tar_files()` (as opposed to the single operation provided in the second argument to `tar_target()` for the `p2_reservoir_ids` target)?

Reply:
Yeah, my instinct says their might be a better way to do this, but I'm not seeing it right now and feel mostly OK with the current implementation.
* `depends <- ` syntax - Yes, the problem I was trying to solve was that I don't want this target to get created until the meteo files have all been made locally available via the 1_fetch step, which is guaranteed by the currentness of `p1_meteo_files`. To get _targets_ to think of `p1_meteo_files` as a dependency, any code that included `p1_meteo_files` as an object or argument would suffice. I think I could also have done `p1_meteo_files` alone on its own line, but I thought assigning it to a variable named `depends` might help in clarifying my intent.
* And yes, another syntax option would have been to write a wrapper function that accepts `p1_meteo_files` as a gratuitous argument...short story is I just didn't/don't feel like it. To me an 8-line code chunk (including the curly braces) isn't that much more inconvenient to read through than a 3-4 line call to a single command, so I kept it all in this file. But I think there's room for our team to develop a consensus opinion about how much code is too much within a _targets_ file, as we get more experience reading and writing these.
* Curly braces - yes, they're there simply because there are multiple commands being passed as a single argument to `tar_files`, so the appropriate R syntax to group them together is `{ }`.